caikit/core/model_management/model_trainer_base.py

```diff
@@ -61,27 +60,26 @@ def __init__(
             self,
             trainer_name: str,
             training_id: str,
-            save_with_id: bool,
```

Comment:
Definitely a breaking API change because it breaks existing trainer implementations